mrp_workorder_ref/models/mrp_workorder.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023 Camptocamp SA (https://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MrpWorkorder(models.Model):
+    _inherit = "mrp.workorder"
+
+    ref = fields.Char(string="Reference")
+
+    _sql_constraints = [
+        (
+            "ref_uniq",
+            "UNIQUE(production_id, ref)",
+            "The Work Order Reference must be unique per MO!",
+        ),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # OVERRIDE to auto-complete the reference from the operation
+        for vals in vals_list:
+            if "ref" not in vals and vals.get("operation_id"):
+                operation_id = vals["operation_id"]
+                operation = self.env["mrp.routing.workcenter"].browse(operation_id)
+                if operation.ref:
+                    vals["ref"] = operation.ref
```

Comment:
IMO would be cleaner to have this in a dedicated function.

Reply:
Hm, I'm not sure I agree. I tend to keep patterns that manipulate `vals` directly in `create` / `write` overrides, with the only exception where they are too complex or shared among them. But in this case it's neither